src/model/Types_6D_Modes.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Types_6D_modes.h
+ *
+ *  Created on: Nov 4, 2017
+ *      Author: glenn
+ */
+
+#ifndef SRC_MODEL_TYPES_6D_MODES_H_
+#define SRC_MODEL_TYPES_6D_MODES_H_
+
+#include "nativeTypesWrapper.h"
+#include "Vec3.h"
+#include "GenericTypes.h"
+
+namespace as {
+#ifndef __CUDACC__ // ostream is not available in nvcc
+template<typename REAL>
+struct DOF_6D_Modes;
+
+template<typename REAL>
+std::ostream& operator<< (std::ostream& s, DOF_6D_Modes<REAL> const&);
+
+template<typename REAL>
+struct Result_6D_Modes;
+
+template<typename REAL>
+std::ostream& operator<< (std::ostream& s, Result_6D_Modes<REAL> const& args);
+
+#endif
+
+template<typename REAL>
+struct DOF_6D_Modes {
+	using real_t = typename TypeWrapper<REAL>::real_t;
+	using vec3_t = Vec3<real_t>;
+	vec3_t pos;
+	vec3_t ang;
+	real_t modes[10];
```

Comment:
better don't hard-code the "10" for the number of modes. For this kind of things I prefer to the create a configuration file (config.h ?!) where you define the maximum allowed number of modes. This file can be included in any modules that need that information. Thus we are more flexible, if we want to change the number later...

Reply:
war mir auch ein bischen unangenehm das so zu machen. Die Lösung kling gut, ich werde es einbauen. 
Eine alternative die mir eingefallen ist, war die Anzahl der Moden beim erstellen einer DOF struct zu initialisieren. Was hälst du davon?